kernel/src/net/socket/netlink/kobject_uevent/message/uevent.rs

```diff
@@ -0,0 +1,186 @@
+// SPDX-License-Identifier: MPL-2.0
+
+// FIXME: The code in this file is currently unused
+// because we haven't implemented support for the uevent inode in sysfs yet.
+#![cfg_attr(not(ktest), expect(dead_code))]
+
+use alloc::format;
+use core::{
+    str::FromStr,
+    sync::atomic::{AtomicU64, Ordering},
+};
+
+use super::syn_uevent::{SyntheticUevent, Uuid};
+use crate::prelude::*;
+
+/// Kobject action type.
+///
+/// Reference: <https://elixir.bootlin.com/linux/v6.14/source/include/linux/kobject.h#L53>.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, TryFromInt)]
+#[repr(u8)]
+pub enum KobjectAction {
+    /// Indicates the addition of a new kobject to the system.
+    ///
+    /// Triggered when a device is discovered or registered.
+    Add = 0,
+
+    /// Signals the removal of a kobject from the system.
+    ///
+    /// Typically occurs during device disconnection or deregistration.
+    Remove = 1,
+
+    /// Denotes a modification to the kobject's properties or state.
+    ///
+    /// Used for attribute changes that don't involve structural modifications.
+    Change = 2,
+
+    /// Represents hierarchical relocation of a kobject.
+    ///
+    /// Occurs when a device moves within the device tree topology.
+    Move = 3,
+
+    /// Marks a device returning to operational status after being offlined.
+    ///
+    /// Common in hot-pluggable device scenarios.
+    Online = 4,
+
+    /// Indicates a device entering non-operational status.
+    ///
+    /// Typically precedes safe removal of hot-pluggable hardware.
+    Offline = 5,
+
+    /// Signifies successful driver-device binding.
+    ///
+    /// Occurs after successful driver probe sequence.
+    Bind = 6,
+
+    /// Indicates driver-device binding termination.
+    ///
+    /// Precedes driver unload or device removal.
+    Unbind = 7,
+}
+
+const KOBJECT_ACTION_STRS: [&str; KobjectAction::Unbind as usize + 1] = [
+    "add", "remove", "change", "move", "online", "offline", "bind", "unbind",
+];
+
+impl FromStr for KobjectAction {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self> {
+        let Some(index) = KOBJECT_ACTION_STRS
+            .iter()
+            .position(|action_str| s == *action_str)
+        else {
+            return_errno_with_message!(Errno::EINVAL, "the string is not a valid kobject action");
+        };
+
+        Ok(KobjectAction::try_from(index as u8).unwrap())
+    }
+}
+
+impl KobjectAction {
+    fn as_str(&self) -> &'static str {
+        KOBJECT_ACTION_STRS[*self as usize]
+    }
+}
+
+/// Kobject userspace event.
+pub struct Uevent {
+    /// The kobject action.
+    action: KobjectAction,
```

Comment:
I remembered that @tatetian mentioned that "kobject" is a bad name and that we shouldn't use it. I guess we end up to use `SysObj` or something similar, but I'm not sure.

In this particular case, the protocol name is "KOBJECT_UEVENT" and we cannot fix the protocol name, but we may want to fix some internal structure names?